ui/src/data/source.tsx

```diff
@@ -531,203 +500,190 @@ function isInternalAttribute(attribute: string): boolean {
-      },
+    underlayName: underlay.name,
+    queryV2: {
+      includeAttributes: normalizeRequestedAttributes(
+        grouping?.attributes ?? requestedAttributes
+      ),
+      includeHierarchyFields:
+        !!classification.hierarchy && !grouping
+          ? {
+              hierarchies: [classification.hierarchy],
+              fields: [
+                tanagra.QueryV2IncludeHierarchyFieldsFieldsEnum.Path,
+                tanagra.QueryV2IncludeHierarchyFieldsFieldsEnum.NumChildren,
+              ],
+            }
+          : undefined,
+      includeRelationshipFields: !grouping
+        ? [
+            {
+              relatedEntity: underlay.primaryEntity,
+              hierarchies: !!classification.hierarchy
+                ? [classification.hierarchy]
+                : undefined,
+            },
+          ]
+        : undefined,
+      filter:
+        makeBooleanLogicFilter(
+          tanagra.BooleanLogicFilterV2OperatorEnum.And,
+          operands
+        ) ?? undefined,
+      orderBys: [makeOrderBy(underlay, classification, grouping)],
     },
   };
   return req;
 }
 
+function convertSortDirection(dir: SortDirection) {
+  return dir === SortDirection.Desc
+    ? tanagra.QueryV2OrderBysDirectionEnum.Descending
+    : tanagra.QueryV2OrderBysDirectionEnum.Ascending;
+}
+
 function processEntitiesResponse(
   attributeID: string,
-  response: tanagra.SearchEntityInstancesResponse,
+  response: tanagra.InstanceListV2,
   grouping?: string
 ): ClassificationNode[] {
-  const pathAttribute = makePathAttribute(attributeID);
-  const numChildrenAttribute = makeNumChildrenAttribute(attributeID);
-
   const nodes: ClassificationNode[] = [];
   if (response.instances) {
     response.instances.forEach((instance) => {
+      const data = processEntityInstance(attributeID, instance);
+
       let ancestors: DataKey[] | undefined;
-      let childCount: number | undefined;
-
-      const data = processEntityInstance(
-        attributeID,
-        instance,
-        (key, value) => {
-          if (key === pathAttribute) {
-            if (value?.stringVal) {
-              // TODO(tjennison): There's no way to tell if the IDs should
-              // be numbers or strings and getting it wrong will cause
-              // queries using them to fail because the types don't match.
-              // Figure out a way to handle this generically.
-              ancestors = value.stringVal.split(".").map((id) => +id);
-            } else if (value?.stringVal === "") {
-              ancestors = [];
-            }
-          } else if (key === numChildrenAttribute) {
-            childCount = value?.int64Val;
-          } else {
-            return false;
-          }
-          return true;
+      const path = instance.hierarchyFields?.[0]?.path;
+      if (isValid(path)) {
+        if (path === "") {
+          ancestors = [];
+        } else {
+          ancestors = path.split(".").map((id) => id as typeof data.key);
```

Comment:
Maybe the backend should return the path as an array of strings instead. Using a `"."` separator is an implementation detail of how we store the path on the backend and may (probably should) change in the future. Thoughts?

Reply:
I agree. An array would be a more robust implementation.